superset/viz.py

```diff
@@ -1224,7 +1224,12 @@ def get_data(self, df):
         comparison_type = fd.get('comparison_type') or 'values'
         df = self.process_data(df)
         if comparison_type == 'values':
-            chart_data = self.to_series(df)
+            # Filter out series with all NaN
```

Comment:
You should be able to do here:

```python
chart_data = self.to_series(df.dropna(axis=1, how='all'))
```

This should be much faster, since it list comprehension are slow, and the `dropna` loop should be in C.